openml/flows/flow.py

```diff
@@ -1,80 +1,358 @@
+    components : OrderedDict
+        Mapping from component identifier to an OpenMLFlow object.
+    parameters : OrderedDict
+        Mapping from parameter name to the parameter default value. The
+        parameter default value must be of type `str`, so that the respective
+        toolbox plugin can take care of casting the parameter default value to
+        the correct type.
+    parameters_meta_info : OrderedDict
+        Mapping from parameter name to `dict`. Stores additional information for
+        each parameter. Required keys are `data_type` and `description`.
+    external_version : str
+        Version number of the software the flow is implemented in. Is used
+        together with the attribute `name` as a uniquer identifier of the flow.
+    uploader : str
+        OpenML user ID of the uploader. Filled in by the server.
+    tags : list
+        List of tags. Created on the server by other API calls.
+    binary_url : str
+        ??? - don't use - implemented because it is used by flows on the server
+    binary_format : str
+        ??? - don't use - implemented because it is used by flows on the server
+    binary_md5 : str
+        ??? - don't use - implemented because it is used by flows on the server
+    version : str
+        OpenML version of the flow.
+    upload_date : str
+        Date the flow was uploaded. Filled in by the server.
+    language : str
+        Natural language the flow is described in (not the programming
+        language).
+    dependencies : str
+        A list of dependencies necessary to run the flow.
     flow_id : int, optional
         Flow ID. Assigned by the server (fixme shouldn't be here?)
-    uploader : string, optional
-        User uploading the model (fixme shouldn't be here?). Assigned by the server.
-
-
     """
-    def __init__(self, model, flow_id=None, uploader=None,
-                 description='Flow generated by run_task', creator=None,
-                 contributor=None, tag=None):
-        self.flow_id = flow_id
-        self.upoader = uploader
+    # TODO @Jan can you find better descriptions for binary_url, binary_md5,
+    # binary_format and version?
+    def __init__(self, name, description=None, model=None, components=None,
+                 parameters=None, parameters_meta_info=None,
+                 external_version=None, uploader=None, tags=None,
+                 binary_url=None, binary_format=None, binary_md5=None,
+                 version=None, upload_date=None, language=None,
+                 dependencies=None, flow_id=None):
+        self.name = name
         self.description = description
-        self.creator = creator
-        self.tag = tag
         self.model = model
-        self.source = "FIXME DEFINE PYTHON FLOW"
-        self.name = (model.__module__ + "." +
-                     model.__class__.__name__)
-        self.external_version = 'sklearn_' + sklearn.__version__
 
-    def _generate_flow_xml(self):
+        if components is None:
```

Comment:
Will create a for loop, type checking still makes sense in my opinion to prevent unnecessary errors.